ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -758,20 +761,26 @@ CompletableFuture<Void> loadFromDisk() {
       return serializeAndDedupOperation(this::executeLoadFromDisk, readInProgress);
     }
 
+    private void cancelInProgressReadOperation() {
+      if (!readInProgress.get().isDone()) {
+        LOG.debug("Cancelling in progress read operation");
+        isCancelled.set(true);
+        try {
+          readInProgress.get().get();
```

Comment:
It is not very clear in the code why this line is cancelling the read operation. Maybe it's worth it to put this call into a method with a descriptive name

Reply:
`readInProgress.get().get()` wait for the completable future to finish, introduced a method to describe its meaning